beacon/validator/src/main/java/tech/pegasys/teku/validator/coordinator/BlockOperationSelectorFactory.java

```diff
@@ -126,6 +126,7 @@ public Function<BeaconBlockBodyBuilder, SafeFuture<Void>> createSelector(
       final SszList<Attestation> attestations =
           attestationPool.getAttestationsForBlock(
               blockSlotState, new AttestationForkChecker(spec, blockSlotState));
+      blockProductionPerformance.getAttestationsForBlock();
```

Comment:
maybe we can add a `beaconBlockPreparationStarted` at the beginning of the bodyBuilder lambda, otherwise we will measure the time from the previous step which is currently `retrieve_state`

Reply:
makes total sense to me, added the preparationstarted event